# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ron/tests/203_error_positions.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/203_error_positions.rs
--- a/third_party/rust/ron/tests/203_error_positions.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/ron/tests/203_error_positions.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -1,11 +1,17 @@
 use std::num::NonZeroU32;
 
-use ron::error::{Error, Position, SpannedError};
+use ron::error::{Error, Position, Span, SpannedError};
 use serde::{
     de::{Deserialize, Error as DeError, Unexpected},
     Deserializer,
 };
 
+#[cfg(feature = "internal-span-substring-test")]
+use ron::util::span_substring::check_error_span_inclusive;
+
+#[cfg(feature = "internal-span-substring-test")]
+use ron::util::span_substring::check_error_span_exclusive;
+
 #[derive(Debug, serde::Deserialize, PartialEq)]
 #[serde(deny_unknown_fields)]
 enum Test {
@@ -24,82 +30,124 @@
 
 #[test]
 fn test_error_positions() {
-    assert_eq!(
-        ron::from_str::<TypeError>("  ()"),
-        Err(SpannedError {
-            code: Error::InvalidValueForType {
-                expected: String::from("impossible"),
-                found: String::from("a unit value"),
-            },
-            position: Position { line: 1, col: 3 },
-        })
-    );
+    let bogus_struct = "  ()";
+    let expected_err = Err(SpannedError {
+        code: Error::InvalidValueForType {
+            expected: String::from("impossible"),
+            found: String::from("a unit value"),
+        },
+        span: Span {
+            start: Position { line: 1, col: 1 },
+            end: Position { line: 1, col: 3 },
+        },
+    });
+
+    assert_eq!(ron::from_str::<TypeError>(bogus_struct), expected_err,);
+
+    #[cfg(feature = "internal-span-substring-test")]
+    check_error_span_inclusive::<TypeError>(bogus_struct, expected_err, "  (");
 
-    assert_eq!(
-        ron::from_str::<Test>("StructVariant(a: true, b: 0, c: -42)"),
-        Err(SpannedError {
-            code: Error::InvalidValueForType {
-                expected: String::from("a nonzero u32"),
-                found: String::from("the unsigned integer `0`"),
-            },
-            position: Position { line: 1, col: 28 },
-        })
-    );
+    let bogus_struct = "StructVariant(a: true, b: 0, c: -42)";
+    let expected_err = Err(SpannedError {
+        code: Error::InvalidValueForType {
+            expected: String::from("a nonzero u32"),
+            found: String::from("the unsigned integer `0`"),
+        },
+        span: Span {
+            start: Position { line: 1, col: 27 },
+            end: Position { line: 1, col: 28 },
+        },
+    });
+
+    assert_eq!(ron::from_str::<Test>(bogus_struct), expected_err);
+
+    #[cfg(feature = "internal-span-substring-test")]
+    check_error_span_inclusive::<Test>(bogus_struct, expected_err, "0,");
 
-    assert_eq!(
-        ron::from_str::<Test>("TupleVariant(42)"),
-        Err(SpannedError {
-            code: Error::ExpectedDifferentLength {
-                expected: String::from("tuple variant Test::TupleVariant with 2 elements"),
-                found: 1,
-            },
-            position: Position { line: 1, col: 16 },
-        })
-    );
+    let bogus_struct = "TupleVariant(42)";
+    let expected_err = Err(SpannedError {
+        code: Error::ExpectedDifferentLength {
+            expected: String::from("tuple variant Test::TupleVariant with 2 elements"),
+            found: 1,
+        },
+        span: Span {
+            start: Position { line: 1, col: 16 },
+            end: Position { line: 1, col: 16 },
+        },
+    });
+
+    assert_eq!(ron::from_str::<Test>(bogus_struct), expected_err);
+
+    #[cfg(feature = "internal-span-substring-test")]
+    check_error_span_inclusive::<Test>(bogus_struct, expected_err, ")");
 
-    assert_eq!(
-        ron::from_str::<Test>("NotAVariant"),
-        Err(SpannedError {
-            code: Error::NoSuchEnumVariant {
-                expected: &["TupleVariant", "StructVariant"],
-                found: String::from("NotAVariant"),
-                outer: Some(String::from("Test")),
-            },
-            position: Position { line: 1, col: 12 },
-        })
-    );
+    let bogus_struct = "NotAVariant";
+    let expected_err = Err(SpannedError {
+        code: Error::NoSuchEnumVariant {
+            expected: &["TupleVariant", "StructVariant"],
+            found: String::from("NotAVariant"),
+            outer: Some(String::from("Test")),
+        },
+        span: Span {
+            start: Position { line: 1, col: 1 },
+            end: Position { line: 1, col: 12 },
+        },
+    });
+
+    assert_eq!(ron::from_str::<Test>(bogus_struct), expected_err);
+
+    #[cfg(feature = "internal-span-substring-test")]
+    check_error_span_exclusive::<Test>(bogus_struct, expected_err, "NotAVariant");
+
+    let bogus_struct = "StructVariant(a: true, b: 1, c: -42, d: \"gotcha\")";
+    let expected_err = Err(SpannedError {
+        code: Error::NoSuchStructField {
+            expected: &["a", "b", "c"],
+            found: String::from("d"),
+            outer: Some(String::from("StructVariant")),
+        },
+        span: Span {
+            start: Position { line: 1, col: 38 },
+            end: Position { line: 1, col: 39 },
+        },
+    });
+
+    assert_eq!(ron::from_str::<Test>(bogus_struct), expected_err);
 
-    assert_eq!(
-        ron::from_str::<Test>("StructVariant(a: true, b: 1, c: -42, d: \"gotcha\")"),
-        Err(SpannedError {
-            code: Error::NoSuchStructField {
-                expected: &["a", "b", "c"],
-                found: String::from("d"),
-                outer: Some(String::from("StructVariant")),
-            },
-            position: Position { line: 1, col: 39 },
-        })
-    );
+    #[cfg(feature = "internal-span-substring-test")]
+    check_error_span_inclusive::<Test>(bogus_struct, expected_err, "d:");
+
+    let bogus_struct = "StructVariant(a: true, c: -42)";
+    let expected_err = Err(SpannedError {
+        code: Error::MissingStructField {
+            field: "b",
+            outer: Some(String::from("StructVariant")),
+        },
+        span: Span {
+            start: Position { line: 1, col: 30 },
+            end: Position { line: 1, col: 30 },
+        },
+    });
+
+    assert_eq!(ron::from_str::<Test>(bogus_struct), expected_err);
 
-    assert_eq!(
-        ron::from_str::<Test>("StructVariant(a: true, c: -42)"),
-        Err(SpannedError {
-            code: Error::MissingStructField {
-                field: "b",
-                outer: Some(String::from("StructVariant")),
-            },
-            position: Position { line: 1, col: 30 },
-        })
-    );
+    #[cfg(feature = "internal-span-substring-test")]
+    check_error_span_inclusive::<Test>(bogus_struct, expected_err, ")");
 
-    assert_eq!(
-        ron::from_str::<Test>("StructVariant(a: true, b: 1, a: false, c: -42)"),
-        Err(SpannedError {
-            code: Error::DuplicateStructField {
-                field: "a",
-                outer: Some(String::from("StructVariant")),
-            },
-            position: Position { line: 1, col: 31 },
-        })
-    );
+    let bogus_struct = "StructVariant(a: true, b: 1, a: false, c: -42)";
+    let expected_err = Err(SpannedError {
+        code: Error::DuplicateStructField {
+            field: "a",
+            outer: Some(String::from("StructVariant")),
+        },
+        span: Span {
+            start: Position { line: 1, col: 30 },
+            end: Position { line: 1, col: 31 },
+        },
+    });
+
+    assert_eq!(ron::from_str::<Test>(bogus_struct), expected_err);
+
+    #[cfg(feature = "internal-span-substring-test")]
+    check_error_span_inclusive::<Test>(bogus_struct, expected_err, "a:");
 }